### PR TITLE
At least for RequestsAdapter, json conversion must be performed

### PR DESCRIPTION
**Pull request by ticosax**
DO NOT MERGE - this is a draft

I suggesting this PR to start a conversation.
It seems this patch is required with RequestsAdapter.
But I suspect it won't work with other adapters.
What do you think ?
Which adapter are you using ?

### CODE REVIEW

**Comment by asyncee** [comment]
Hey, thanks for suggesting that feature, did you hit some erroneous situation or maybe some specific schema/remote endpoint that requires such a patch? RequestsAdapter uses `json` keyword argument while calling `requests`, so using `dict()` instead of `json.dumps()` is fine.

**Comment by ticosax** [comment]
Yes, it broke for me with date field.
I will try to write a reproducible test

**Comment by asyncee** [comment]
Did you get the `TypeError: Object of type datetime is not JSON serializable`? There are few solutions to that, but it is better to wait for reproducible test. Thanks!

File: tests/test_response_deserializer.py [hunk 1]
```diff
@@ -17,26 +20,40 @@ def d() -> ResponseDeserializer:
...
     assert d.deserialize(set, {1, 2, 3}) == {1, 2, 3}
+
+
+def test_pydantic_serializer():
+    date = datetime.date(2020, 12, 1)
+    resource = PydanticClass(field="hello", date=date)
+    assert resource.dict() == {"field": "hello", "date": date}
+    assert resource.json() == json.dumps({"field": "hello", "date": "2020-12-01"})
+    with pytest.raises(TypeError) as execinfo:
+        json.dumps(resource.dict())
```
Comment (ticosax) [inline tests/test_response_deserializer.py:58]: If we let the adapter perform the json conversion, it is not working. Hence the proposal to let pydantic perform conversion.

**Comment by asyncee** [comment]
I have fixed it slightly differently in [custom-json-serialization](https://github.com/asyncee/swagger_codegen/tree/custom-json-serialization) branch.

Can you confirm if the proposed changes work for you?

**Comment by ticosax** [comment]
Nice one!
Works like a charm,
Just a minor thing, I had to restore import of `datetime` though
https://github.com/ticosax/swagger_codegen/tree/custom-json-serialization

Thanks a lot!

**Comment by asyncee** [comment]
Thanks, i missed that import. I had to implement simple integration tests to prevent such issues in future.

Version 0.1.30 released and includes all the changes.

**Comment by asyncee** [comment]
Also, @ticosax thank you for your help and time! ;)

**Pull request closed without merging**